sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1188,6 +1188,12 @@ class Dataset[T] private[sql](
   def col(colName: String): Column = colName match {
     case "*" =>
       Column(ResolvedStar(queryExecution.analyzed.output))
+    case ParserUtils.escapedIdentifier(columnNameRegex)
```

Comment:
@cloud-fan I might have misunderstood your last comment "I don't think sql a-like syntax is really useful here. How about we create a special cased col function that takes a regex?". Can you clarify with some examples? Thanks.